### ISSUE TITLE
New warning in Ruby 3.2.0

### ISSUE THREAD

**Issue by rafaelmagu** [tonytonyjan/exif#32]
With the latest (3.2.0 as of this writing) version of Ruby, the gem yields a warning when instantiating a new object.

Given the following code:
```ruby
require 'exif'

# Instantiates an Exif::Data object from tempfile
data = Exif::Data.new(File.open(file))

# Returns EXIF data in hash format, if present
if data.to_h.key?(:gps) && data.to_h.key?(:exif)
  puts data.to_h
else
  logger.info 'Skipping image analysis because no Exif or GPS information exists'
end
```

Ruby 3.2.0 prints out the following on stderr:

`warning: undefining the allocator of T_DATA class Exif::Data` for line 4 (the `Exif::Data.new(...)` bit).

The same code yielded no warnings on Ruby 2.7.5.

A similar issue discovered and fixed in another project: https://github.com/vmg/redcarpet/pull/721/files#diff-94e6d4cf98d0a12f376b9c3fb158dbc2ada0c98d2d230d356d7b7fd467a80619R179

**Comment by tonytonyjan** [comment]
Thanks @rafaelmagu! I have pushed a fix to [fix/undefine-unnecessary-allocator](https://github.com/tonytonyjan/exif/tree/fix/undefine-unnecessary-allocator) can you help me double check if it really eliminates the warning and breaks nothing? Thanks.

**Comment by rafaelmagu** [comment]
@tonytonyjan I can confirm the fix in that branch works as expected in 3.2.0.

To test, I changed my `Gemfile` from

```ruby
gem 'exif'
```
to
```ruby
gem 'exif', git: 'https://github.com/tonytonyjan/exif.git', branch: 'fix/undefine-unnecessary-allocator'
```

And ran my Rspec suite of tests. No warning appeared, where before it was the first piece of output.

**Comment by tonytonyjan** [comment]
@rafaelmagu thanks for making this gem better!
